logstreams/src/main/java/io/zeebe/logstreams/impl/log/fs/FsLogStorage.java

```diff
@@ -325,7 +326,10 @@ public void flush() throws Exception {
 
     if (dirtySegmentId >= 0) {
       for (int id = dirtySegmentId; id <= currentSegment.getSegmentId(); id++) {
-        logSegments.getSegment(id).flush();
+        final FsLogSegment segment = logSegments.getSegment(id);
+        if (segment != null) {
```

Comment:
With this I'm bit unsure, maybe we should log that this was Null? Because with this null check we will not see that we have an bug here :sweat_smile: